compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -1263,7 +1263,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 let InferOk { obligations, .. } = self
                     .infcx
                     .at(&obligation.cause, obligation.param_env)
-                    .eq(DefineOpaqueTypes::No, b, a)
+                    .eq(DefineOpaqueTypes::Yes, b, a)
```

Comment:
Doesn't this cause the following to pass:

```
fn hello() -> [impl Sized; 2] {
    if false {
        let x = hello();
        let _: &[i32] = x;
    }
    todo!()
}
```

Reply:
yea, I removed those changes now